ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts

```diff
@@ -0,0 +1,285 @@
+    await moveMouseToTheMiddleOfTheScreen(page);
+    const { x, y } = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await dragMouseTo(x, y + delta, page);
+    await page.mouse.click(x, y + delta);
+  });
+
+  test('Click or drag on the canvas: Rotate with a benzene', async ({
+    page,
+  }) => {
+    /*
+      Test case: 1678
+      Description: Repeat the previous steps with different templates.
+      */
+    await selectRing(RingButton.Benzene, page);
+    await moveMouseToTheMiddleOfTheScreen(page);
+    await page.keyboard.down('Control');
+    const { x, y } = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await clickInTheMiddleOfTheScreen;
+    await page.mouse.move(x + delta, y + delta);
+    await dragMouseTo(x + delta + delta, y, page);
+    await page.keyboard.up('Control');
+  });
+
+  test('Undo/Redo action', async ({ page }) => {
+    /*
+      Test case: 1737
+      Description: Edit the structures in any way.
+                   Click Undo multiple times.
+                   Click Redo multiple times.
+      */
+    await openFileAndAddToCanvas('template-manipulations/TM-1737.mol', page);
+    await clickInTheMiddleOfTheScreen(page);
+    point = await getTopAtomByAttributes(page, {
+      label: 'O',
+    });
+    await page.mouse.click(point.x, point.y);
+    await dragMouseTo(point.x + delta, point.y + delta, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+  });
+
+  test('Rotate/Flip the template structure', async ({ page }) => {
+    /*
+            Test case: 1745
+            Description: Choose the 'Rotate Tool', select the structure and rotate it.
+                Select the structure and flip it horizontally with the 'Horizontal Flip' tool.
+                Select the structure and flip it vertically with the 'Vertical Flip' tool.
+        */
+    await openFileAndAddToCanvas('template-manipulations/TM-1737.mol', page);
+    await clickInTheMiddleOfTheScreen(page);
+    await page.keyboard.press('Control+a');
+    await pressButton(page, 'Vertical Flip (Alt+V)');
+    await pressButton(page, 'Horizontal Flip (Alt+H)');
+  });
+
+  test('Save as *.mol file', async ({ page }) => {
+    /*
+        Test case: 1747
+        Description: Click the 'Save As' button, and click the 'Save' button.
+        Open the saved *.mol file and edit it in any way.
+      */
+    await selectRing(RingButton.Benzene, page);
+    const expectedFile = await getMolfile(page, 'v2000');
+    await saveToFile('TM-1737.mol', expectedFile);
+    const METADATA_STRING_INDEX = [1];
+    const { fileExpected: molFileExpected, file: molFile } =
+      await receiveFileComparisonData({
+        page,
+        expectedFileName: 'tests/test-data/template-manipulations/TM-1737.mol',
```

Comment:
Could you please give meaningful filenames? `TM-1737.mol` doesn't make any sence